### DISCUSSION TITLE
"NextRouter was not mounted" with a twist

### DISCUSSION THREAD

**Discussion by ackvf** [vercel/next.js#66836] [Help]
### Summary

Hello, my repository is corrupted and I don't know how ti fix it. I am getting this error, despite it making no sense, because it worked an our ago and everyone else in the team says it's working for them on my branches too. Are there any other causes that may produce this error in addittion to those listed at the [link](https://nextjs.org/docs/messages/next-router-not-mounted)? Any ideas how to solve this?

```
Error: NextRouter was not mounted. https://nextjs.org/docs/messages/next-router-not-mounted
```

#### What I did that lead to this:

I was in charge of two things this sprint.
- Upgrade all packages to latest minor
- Transition from `yarn` to `pnpm`

After a lot of hassle and complications I managed to complete both, however, I am now unable to build our project. When I revert to a previous branch, or even a month old branch, that I am 100% sure that worked for me, it still fails with the same error.

Steps I did:
- clear npm cache
- clear yarn cache
- clear pnpm cache
- clear next cache
- delete node_modules
- reinstall dependencies
- checkout fresh repo to another directory

Nothing worked and I still get the same error.

Thank you

### Additional information

We are using the pages router.

```
next@14.1.1
node@v20.11.1
yarn@1.22.21
```

/edit

I was able to successfully build a new clone in WSL on the same computer, just to verify that the repo is indeed correct. I want to work natively in Windows, though, where it still **fails**.

**Comment by iris-namrata** [comment]
Were you able to solve this issue? Getting same and unable to resolve after upgrading next js 
@leerob can you please help.

**Comment by uttampun44** [comment]
@iris-namrata  https://nextjs.org/docs/messages/next-router-not-mounted
If used in the app directory, migrate to the new hooks imported from next/navigation.
